### PR TITLE
Draft: create update password screen

### PR DESCRIPTION
**Pull request by Cristuker**
I'm working on issue #5

### CODE REVIEW

**Comment by Cristuker** [comment]
This feature is ready for review

File: pages/update-password.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                                type="password"
+                                label="Confirm password"
+                                name="confirmationPassword"
+                                placeholder="Confirm your password"
+                                value={formik.values.confirmationPassword}
+                                error={formik.touched.confirmationPassword ? formik.errors.confirmationPassword : undefined}
+                                onChange={formik.handleChange}
+                            />
+                        </div>
+                        <Card.Footer>
```
Comment (devkiran) [inline pages/update-password.tsx:92]: Move the `<Card.Footer>..</Card.Footer>` outside the `<Card.Body>`.

File: pages/update-password.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        },
+        validationSchema: Yup.object().shape({
+            currentPassword: Yup.string().required(),
+            newPassword: Yup.string().required('Required'),
+            confirmationPassword: Yup.string().oneOf([Yup.ref('newPassword'), null], "Passwords don't match!")
+            .required('Required')
+        }),
+        onSubmit: async (values) => {
+            const { confirmationPassword } = values;
+            const currentPasswordIsValid = await verifyPassword(formik.values.currentPassword, String(user?.password));
```
Comment (devkiran) [inline pages/update-password.tsx:34]: Should we do `verifyPassword` on the server side? Send all the 3 form values to the `/api/users` and do all the logic on the API route and display error message on the UI.

File: pages/update-password.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import { inferSSRProps } from "@/lib/inferSSRProps";
```
Comment (devkiran) [inline pages/update-password.tsx:1]: Spacing and indentation should be consistent throughout the codebase. We use the 2-space indentation I think.

**Review by devkiran — CHANGES_REQUESTED** [review]

File: pages/update-password.tsx (hunk 3 above)
Comment (Cristuker) [reply pages/update-password.tsx:1]: Ok, I run `npm run format` I think this is enough

**Comment by devkiran** [comment]
@Cristuker Make sure you re-request review after you make the changes so that we know it is ready to review again.

File: .eslintcache [hunk 4]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+[{"/home/tucks/github/saas-starter-kit/pages/api/users.ts":"1"},{"size":1398,"mtime":1665753553349,"results":"2","hashOfConfig":"3"},{"filePath":"4","messages":"5","suppressedMessages":"6","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1v3o9gp","/home/tucks/github/saas-starter-kit/pages/api/users.ts",[],[]]
```
Comment (niwsa) [inline .eslintcache:1]: This file need not be checked in. Can be added to `.gitignore`.

**Comment by Cristuker** [comment]
@devkiran I only request for revire after make changes, don't worry. @niwsa PR is ready for review again.

**Comment by niwsa** [comment]
@Cristuker It would be nice to have the 'Update Password' to be nested under 'Account'. For reference it could be similar to the github settings page https://github.com/settings. So in future, more fields could come under account.

**Comment by Cristuker** [comment]
I make the changes you request, can you review again?

> @Cristuker It would be nice to have the 'Update Password' to be nested under 'Account'. For reference it could be similar to the github settings page https://github.com/settings. So in future, more fields could come under account.

**Comment by niwsa** [comment]
Looks good to me.

**Review by devkiran — APPROVED** [review]

**Comment by devkiran** [comment]
Thank you for your time and this contribution.

**Pull request merged**